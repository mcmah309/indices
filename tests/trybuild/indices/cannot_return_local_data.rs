use indices::indices;

fn main() {
    one();
    two();
    three();
    four();
    five();
}

fn one() -> &'static mut i32 {
    let mut data = [5, 4, 3, 2, 1];
    let slice = data.as_mut_slice();
    let (four,) = indices!(slice, 4);
    four
}

fn two() -> (&'static mut i32, &'static mut i32) {
    let mut data = [5, 4, 3, 2, 1];
    let slice = data.as_mut_slice();
    let (one, four) = indices!(slice, 1, 4);
    (one, four)
}

fn three() -> &'static mut i32 {
    let mut data = [5, 4, 3, 2, 1];
    let slice = data.as_mut_slice();
    let (one, _two, _three) = indices!(slice, 1, 2, 3);
    one
}

fn four() -> (&'static mut i32, &'static mut i32, &'static mut i32, &'static mut i32) {
    let mut data = [5, 4, 3, 2, 1];
    let slice = data.as_mut_slice();
    let (one, two, three, four) = indices!(slice, 1, 2, 3, 4);
    (one, two, three, four)
}

fn five() -> (&'static mut i32, &'static mut i32, &'static mut i32, &'static mut i32, &'static mut i32) {
    let mut data = [5, 4, 3, 2, 1];
    let slice = data.as_mut_slice();
    let (zero, one, two, three, four) = indices!(slice, 0, 1, 2, 3, 4);
    (zero, one, two, three, four)
}
