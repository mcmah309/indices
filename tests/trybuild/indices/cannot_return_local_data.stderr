error[E0515]: cannot return value referencing local variable `data`
  --> tests/trybuild/indices/cannot_return_local_data.rs:11:12
   |
9  |     let slice = data.as_mut_slice();
   |                 ---- `data` is borrowed here
10 |     let (one, _two, _three) = indices!(slice, 1, 2, 3);
11 |     return one
   |            ^^^ returns a value referencing data owned by the current function
